### PR TITLE
docs: add docs page for prerendering

### LINKED ISSUE #22011: [Documentation Request] Prerender section
## Details

There's enough here that is unique about Nuxt that I think it makes sense to have in our docs, and we can link out to Nitro where it makes sense

_Originally posted by @danielroe in https://github.com/nuxt/nuxt/issues/21987#issuecomment-1624835848_

### PR DESCRIPTION
**Pull request by RyanClementsHax**
### 🔗 Linked issue

Resolves #22011

### CODE REVIEW

**Comment by RyanClementsHax** [comment]
@danielroe I had a few follow up questions related to [this comment](https://github.com/nuxt/nuxt/issues/22011#issuecomment-2098474200).

- Are there any examples of this? "mention 200.html + 404.html and the need to set these up as fallbacks (e.g. on S3)"
- Is there a list of features to menation here? "mention static builds and providers that support it with extra features (like vercel, netlify, amplify + cloudflare pages)"
- Do you know off the top of your head what takes precedence with all of the different ways to configure prerendering?

I'll happily make any changes you would like for you. This is just the first draft.

File: docs/1.getting-started/9.prerendering.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+Lastly, you can manually configure this using routeRules.
+
+```ts twoslash [nuxt.config.ts]
+export default defineNuxtConfig({
+  routeRules: {
+    // Set prerender to true to configure it to be prerendered
+    "/rss.xml": { prerender: true },
+    // Set it to false to configure it to be skipped for prerendering
+    "/this-DOES-NOT-get-prerendered": { prerender: true },
```
Comment (danielroe) [inline docs/1.getting-started/9.prerendering.md:78]:
```suggestion
    "/this-DOES-NOT-get-prerendered": { prerender: false },
```

File: docs/1.getting-started/9.prerendering.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    // Everything under /blog gets prerendered
```
Comment (danielroe) [inline docs/1.getting-started/9.prerendering.md:79]:
```suggestion
    // Everything under /blog gets prerendered as long as it
    // is linked to from another page
```

File: docs/1.getting-started/9.prerendering.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    "/blog/**": { prerender: true },
+  },
+});
+```
+
+::read-more{to="https://nitro.unjs.io/config/#routerules"}
+Read more about Nitro's `routeRules` configuration.
+::
+
+As a shorthand, can also configure this in a page file using [`defineRouteRules`](/docs/api/utils/define-route-rules).
```
Comment (danielroe) [inline docs/1.getting-started/9.prerendering.md:89]:
```suggestion
As a shorthand, you can also configure this in a page file using [`defineRouteRules`](/docs/api/utils/define-route-rules).
```

File: docs/1.getting-started/9.prerendering.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+export default defineNuxtConfig({
+  routeRules: {
+    "/": { prerender: true },
+  },
+});
+```
+
+## Runtime prerender configuration
+
+### prerenderRoutes
```
Comment (danielroe) [inline docs/1.getting-started/9.prerendering.md:123]:
```suggestion
### `prerenderRoutes`
```

File: docs/1.getting-started/9.prerendering.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+<template>
+  <div>
+    <h1>This will register other routes for prerendering when prerendered</h1>
+  </div>
+</template>
+```
+
+:read-more{title="prerenderRoutes" to="/docs/api/utils/prerender-routes"}
+
+### prerender:routes Nuxt hook
```
Comment (danielroe) [inline docs/1.getting-started/9.prerendering.md:141]:
```suggestion
### `prerender:routes` Nuxt hook
```

File: docs/1.getting-started/9.prerendering.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+      );
+      for (const page of pages) {
+        ctx.routes.add(`/${page.name}`);
+      }
+    },
+  },
+});
+```
+
+### prerender:generate Nitro hook
```
Comment (danielroe) [inline docs/1.getting-started/9.prerendering.md:160]:
```suggestion
### `prerender:generate` Nitro hook
```

**Review by danielroe — COMMENTED** [review]
This is really nice - thank you! ❤️

**Comment by Atinux** [comment]
I love it, thank you ❤️

**Comment by RyanClementsHax** [comment]
@danielroe @Atinux thanks for your kind words :)

Any other changes you would like to see?

**Review by danielroe — APPROVED** [review]

**Comment by RyanClementsHax** [comment]
@danielroe Thanks for taking my contribution! I think there were still some open items you wanted in https://github.com/nuxt/nuxt/issues/22011#issuecomment-2098474200. Would you like me to create a new PR for those? I just [need some clarification what you meant](https://github.com/nuxt/nuxt/pull/27286#issuecomment-2121363590).

**Pull request merged**